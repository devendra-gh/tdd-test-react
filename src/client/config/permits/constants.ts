import { IVariables } from '@tamm/app-composer';
import { PATH_ACCOUNT_UPGRADE, PATH_LOGIN } from './utils/constants/pageRoutes';

export const PROCESS_NAME = 'permits';

export const errorMsgMap: { [key: string]: string } = {
  'Incorrect emirates ID': 'errorMsg.incorrectemiratesid',
  'Incorrect Service Name': 'errorMsg.incorrectservicename',
  'Failed to send message': 'errorMsg.failedtosendmessage',
  'Incorrect DED license number': 'errorMsg.incorrectdedlicensenumber',
  'Incorrect pro mobile number': 'errorMsg.incorrectpromobilenumber',
  'Incorrect pro name': 'errorMsg.incorrectproname',
  'Incorrect email': 'errorMsg.incorrectemail',
  'Incorrect documents': 'errorMsg.incorrectdocuments',
  'Incorrect contact': 'errorMsg.incorrectcontact',
  'Incorrect Advertisement Details': 'errorMsg.incorrectadvertisementdetails',
  'Incorrect Advertisement Location': 'errorMsg.incorrectadvertisementlocation',
  'Incorrect Advertise Start Date': 'errorMsg.incorrectadvertisestartdate',
  'Incorrect Advertise End Date': 'errorMsg.incorrectadvertiseenddate',
  'Incorrect Customer Comment': 'errorMsg.incorrectcustomercomment',
  'Incorrect Study Specialization': 'errorMsg.incorrectstudyspecialization',
  'Incorrect Project Name': 'errorMsg.incorrectprojectname',
  'Incorrect Comments': 'errorMsg.incorrectcomments',
  'Incorrect Quantity': 'errorMsg.incorrectquantity',
  'Incorrect Plate Or Chassis Number': 'errorMsg.incorrectplateorchassisnumber',
  'Incorrect Seasonal Promotions': 'errorMsg.incorrectseasonalpromotions',
  'Incorrect Promotion Details': 'errorMsg.incorrectpromotiondetails',
  'Incorrect Promotion Location': 'errorMsg.incorrectpromotionlocation',
  'Incorrect Advertisements Quantity':
    'errorMsg.incorrectadvertisementsquantity',
  'Incorrect Plate Category': 'errorMsg.incorrectplatecategory',
  'Incorrect Advertisement Type': 'errorMsg.incorrectadvertisementtype',
  'Incorrect Note': 'errorMsg.incorrectnote',
  'Incorrect City': 'errorMsg.incorrectcity',
  'Incorrect Employee Comment': 'errorMsg.incorrectemployeecomment',
  'Incorrect Permit Type': 'errorMsg.incorrectpermittype',
  'Incorrect Airs': 'errorMsg.incorrectairs',
  'Incorrect Sign Boards': 'errorMsg.incorrectsignboards',
  'Please provide machines': 'errorMsg.pleaseprovidemachines',
  'Incorrect Umbrellas': 'errorMsg.incorrectumbrellas',
  'Incorrect Balloons': 'errorMsg.incorrectballoons',
  'Incorrect Sale Banners': 'errorMsg.incorrectsalebanners',
  'Incorrect Prizes': 'errorMsg.incorrectprizes',
  'Incorrect Propaganda Advertising Boards':
    'errorMsg.incorrectpropagandaadvertisingboards',
  'Incorrect Kiosks': 'errorMsg.incorrectkiosks',
  'Incorrect Vehicles': 'errorMsg.incorrectvehicles',
};

export const REQUIRES_SOP3 = {
  test: (props: IVariables) => {
    const { user } = props;
    if (user.Type === 'SOP3') return true;
    return false;
  },
  redirectTo: PATH_ACCOUNT_UPGRADE,
};

export const REQUIRES_CUSTOM_LOGIN = {
  test: (props: IVariables) => {
    if (props.loggedIn) return true;
    return false;
  },
  redirectTo: `${PATH_LOGIN}?redirectUrl=${window.location.href}`,
};
