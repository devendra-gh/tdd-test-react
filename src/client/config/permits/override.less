@import '~@tamm/ui-lib-v2-styles/colors.less';
@import '~@tamm/ui-lib-v2-styles/common';

.ui-lib-form__controls {
  margin-top: 4rem;
  padding-top: 4rem !important;
}

.ui-lib-grid-row {
  padding: 0 !important;
}

.all-icon-close {
  float: right;
}

.ui-lib-notice__icon {
  font-size: 48px !important;
}

.ui-lib-alert {
  align-items: center !important;
}

.ui-lib-checkbox__box {
  margin-top: 0.2rem;
}

.ui-lib-checkbox__label-text {
  font-family: @ui-lib-font-family-roboto, @ui-lib-font-family-noto-kufi;
  color: @ui-lib-body-text-color;
  letter-spacing: normal;
  line-height: 2.2rem !important;
  flex: 1;
}

.ui-lib-pagination {
  justify-content: flex-start !important;
}

.ui-lib-form__fieldset-content {
  margin-bottom: 0 !important;
}

.ui-lib-text-area_small .ui-lib-text-area__content {
  font-size: 1.6rem !important;
}

.ui-lib-form__fieldset,
.ui-lib-viewport_xl .ui-lib-form__fieldset.ui-lib-viewport_lg .ui-lib-form__fieldset {
  margin-bottom: 0 !important;
}
