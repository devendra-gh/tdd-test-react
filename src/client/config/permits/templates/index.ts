import Header from 'client/templates/PermitsHeader';
import Footer from 'client/templates/Footer';

const templates = {
  notice: () => import('./Notice'),
  home: () => import('./Home'),
  login: () => import('./Login'),
  smartpassLogin: () => import('./SmartpassLogin'),
  profile: () => import('./Profile'),
  summary: () => import('./Summary'),
  undertaking: () => import('./Undertaking'),
  permitForm: () => import('./PermitForm'),
  welcome: () => import('./Welcome'),
  page404: () => import('./NotFound'),
  accountUpgrade: () => import('./AccountUpgrade'),
  entityApprovedDocs: () => import('./EntityApprovedDocs'),
  header: Header,
  footer: Footer,
};

export default templates;
