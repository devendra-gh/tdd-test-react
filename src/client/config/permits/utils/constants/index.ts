import { countries } from './countries';
import { legalForms } from './legalForms';
import { companies } from './companies';
import { emirates } from './emirates';
import { nationalities } from './nationalities';
import { permitType } from './permitType';
import { representativeType } from './representativeType';

export {
  companies,
  countries,
  emirates,
  legalForms,
  nationalities,
  permitType,
  representativeType,
};
