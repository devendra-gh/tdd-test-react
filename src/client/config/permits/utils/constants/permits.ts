const PERMIT_ADDITIONAL_SIGNBOARD = 'additional-signboard-permit-in-abu-dhabi';
const PERMIT_PROPAGANDA_AD_BOARD = 'propaganda-ad-board-permit-in-abu-dhabi';
const PERMIT_AWNINGS_AD = 'awnings-ad-permit-in-abu-dhabi';
const PERMIT_FIXED_AD_SIGNBOARD = 'fixed-ad-signboard-permit-in-abu-dhabi';
const PERMIT_BANNER_AD = 'banner-ad-permit-in-abu-dhabi';
const PERMIT_ELECTRONIC_SIGNBOARD = 'electronic-signboard-permit-in-abu-dhabi';
const PERMIT_TEMP_KIOSK_AD = 'temporary-kiosk-ad-permit-in-abu-dhabi';
const PERMIT_AIR_AD = 'air-ad-permit-in-abu-dhabi';
const PERMIT_BALLOONS_AD = 'balloons-ad-permit-in-abu-dhabi';
const PERMIT_ATM_AD = 'atm-ad-permit-in-abu-dhabi';
const PERMIT_CAFE = 'cafe-permit-in-abu-dhabi';
const PERMIT_CLEARANCE = 'clearance-permit-in-abu-dhabi';
const PERMIT_DRAWS = 'draws-permit-in-abu-dhabi';
const PERMIT_VEHICLES_AD = 'vehicles-ad-permit-in-abu-dhabi';
const PERMIT_VENDING_MACHINE = 'vending-machine-ad-permit-in-abu-dhabi';
const PERMIT_PRIZE_DISPLAY_FORMAT =
  'prize-display-platform-permit-in-abu-dhabi';
const PERMIT_PAPER_AD = 'paper-ad-permit-in-abu-dhabi';
const PERMIT_PAPER_PUBLISHING_AD = 'paper-publishing-ad-permit-in-abu-dhabi';
const PERMIT_VARIOUS_MACHINES_AD = 'various-machines-ad-permit-in-abu-dhabi';
const PERMIT_SEASONAL_PROMOTION = 'seasonal-promotions-permit-in-abu-dhabi';
const PERMIT_AUTOMATIC_PAYMENT_AD = 'automatic-payment-ad-permit-in-abu-dhabi';
const PERMIT_SALES = 'sales-permit-in-abu-dhabi';
const PERMIT_FOOD_TRUCK = 'food-truck-permit-in-abu-dhabi';
const PERMIT_MOBILE_CAR = 'mobile-car-tajer-permit-in-abu-dhabi';

export {
  PERMIT_ADDITIONAL_SIGNBOARD,
  PERMIT_PROPAGANDA_AD_BOARD,
  PERMIT_AWNINGS_AD,
  PERMIT_FIXED_AD_SIGNBOARD,
  PERMIT_BANNER_AD,
  PERMIT_ELECTRONIC_SIGNBOARD,
  PERMIT_AIR_AD,
  PERMIT_BALLOONS_AD,
  PERMIT_ATM_AD,
  PERMIT_CAFE,
  PERMIT_VENDING_MACHINE,
  PERMIT_VEHICLES_AD,
  PERMIT_PRIZE_DISPLAY_FORMAT,
  PERMIT_PAPER_AD,
  PERMIT_CLEARANCE,
  PERMIT_DRAWS,
  PERMIT_VARIOUS_MACHINES_AD,
  PERMIT_TEMP_KIOSK_AD,
  PERMIT_SEASONAL_PROMOTION,
  PERMIT_AUTOMATIC_PAYMENT_AD,
  PERMIT_PAPER_PUBLISHING_AD,
  PERMIT_SALES,
  PERMIT_FOOD_TRUCK,
  PERMIT_MOBILE_CAR,
};
