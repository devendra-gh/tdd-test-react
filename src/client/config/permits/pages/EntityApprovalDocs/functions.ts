import { IVariables } from '@tamm/app-composer';
import bpm from 'client/services/bpm';
import {
  PATH_ENTITY_APPROVAL,
  PATH_APPLICATION_WAITING,
} from '../../utils/constants/pageRoutes';
import { PROCESS_NAME } from '../../constants';
import validation from './functions/validation';

const onBackClick = (props: IVariables) => {
  const { serviceType } = props;
  props.history.push(`/${serviceType}${PATH_ENTITY_APPROVAL}`);
};
const onSubmit = async (props: IVariables) => {
  const { businessKey, serviceType, permitInfo } = props;
  const { entityApproval } = permitInfo[serviceType];
  if (Object.keys(entityApproval.documents).length !== 0) {
    await bpm.message(PROCESS_NAME, {
      businessKey,
      messageName: 'onOtherEntitySubmit',
      variables: {
        otherEntitySubmitted: true,
        entityApprovalDocs: JSON.stringify(
          Object.values(
            permitInfo[serviceType].entityApproval.documents,
          ).reduce((fileArray: any, file: any) => fileArray.concat(file), []),
        ),
      },
    });
    props.history.push(`/${serviceType}${PATH_APPLICATION_WAITING}`);
  }
};

const handleToggleCheckbox = (props: IVariables) => {
  const { serviceType, permitInfo } = props;
  props.actions.permitInfo.update({
    ...permitInfo,
    [serviceType]: {
      ...permitInfo[serviceType],
      entityApproval: {
        ...permitInfo[serviceType].entityApproval,
        isApproved: !permitInfo[serviceType].entityApproval.isApproved,
      },
    },
  });
};
export default {
  onBackClick,
  onSubmit,
  validation,
  handleToggleCheckbox,
};
