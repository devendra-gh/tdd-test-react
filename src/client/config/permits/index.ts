import fetchState from './state';
import templates from './templates';
import translations from './localization';
import {
  PATH_UNDERTAKING,
  PATH_APPLICATION_DETAILS,
  PATH_CONTINUE_PROCESS,
  PATH_ENTITY_APPROVAL_DOCS,
} from './utils/constants/pageRoutes';

const skipFetchState: string[] = [
  '/',
  `/:serviceName${PATH_UNDERTAKING}`,
  `/:serviceName${PATH_APPLICATION_DETAILS}`,
  `/:serviceName${PATH_ENTITY_APPROVAL_DOCS}`,
  PATH_CONTINUE_PROCESS,
];

export { fetchState, templates, skipFetchState, translations };
