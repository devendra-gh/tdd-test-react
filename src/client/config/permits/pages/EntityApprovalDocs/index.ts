import { REQUIRES_LOGIN, IVariables } from '@tamm/app-composer';
import { PROCESS_NAME } from 'client/config/permits/constants';
import functions from './functions';
import {
  PATH_ENTITY_APPROVAL_DOCS,
  PATH_APPLICATION_WAITING,
} from '../../utils/constants/pageRoutes';
import getSteps from '../../utils/getSteps';
import {
  STEP_TRACKER_STATUS_FINISH,
  STEP_TRACKER_STATUS_PROCESS,
} from '../../utils/constants/stepTrackerStatus';
import {
  PERMIT_MOBILE_CAR,
  PERMIT_FOOD_TRUCK,
} from '../../utils/constants/permits';
import { STEP_4_1_1 } from '../../steps';

const entityApproval = [
  {
    path: [
      `/${PERMIT_MOBILE_CAR}${PATH_ENTITY_APPROVAL_DOCS}`,
      `/${PERMIT_FOOD_TRUCK}${PATH_ENTITY_APPROVAL_DOCS}`,
    ], // path for router
    uniqueId: 'permits-entityApproval-docs', // uniqueId for caching and other purposes
    template: 'entityApprovedDocs', // template name, must be located in index of folder template/index
    title: 'pageTitle.permitApplication', // title of the page, later it will be read from CMS
    props: {
      buttons: [],
      title: 'entityApproval.title',
      content: 'entityApproval.content',
      onBackClick: functions.onBackClick,
      onSubmit: functions.onSubmit,
      validation: functions.validation,
      handleToggleCheckbox: functions.handleToggleCheckbox,
    },
    onPageInit: (props: IVariables) => {
      const { serviceType, otherEntitySubmitted, steps } = props;
      if (otherEntitySubmitted) {
        props.history.push(`/${serviceType}${PATH_APPLICATION_WAITING}`);
      }
      return {
        currentStep: steps[3].name,
        stepsStatus: {
          [`${steps[3].name}.${STEP_4_1_1}`]: STEP_TRACKER_STATUS_PROCESS,
          [steps[2].name]: STEP_TRACKER_STATUS_FINISH,
          [steps[1].name]: STEP_TRACKER_STATUS_FINISH,
          [steps[0].name]: STEP_TRACKER_STATUS_FINISH,
        },
      };
    },
    fromProcessState: {
      processName: PROCESS_NAME,
      variables: ['otherEntitySubmitted'],
    },
    state: {
      mapState: [
        'loggedIn',
        'stepsStatus',
        {
          steps: (state: IVariables) => {
            return getSteps(state);
          },
        },
        'businessKey',
        'serviceType',
        'permitType',
        'permitInfo',
        'history',
        'user',
      ],
      mapDispatch: ['instanceId', 'businessKey', 'permitInfo'],
    },
    requires: [REQUIRES_LOGIN],
  },
];

export default entityApproval;
