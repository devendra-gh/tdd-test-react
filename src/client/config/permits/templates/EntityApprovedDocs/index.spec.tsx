import { IVariables } from '@tamm/app-composer';
import { shallow } from 'enzyme';
import React, { FunctionComponent } from 'react';
import EntityApprovalDocs from './index';
import { PERMIT_MOBILE_CAR } from '../../utils/constants/permits';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: FunctionComponent) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('permits/templates/EntityApprovalDocs', () => {
  let props: IVariables;
  let e: any;
  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      currentStep: 'test',
      serviceType: PERMIT_MOBILE_CAR,
      onBackClick: jest.fn(),
      onSubmit: jest.fn(),
      handleToggleCheckbox: jest.fn(),
      validation: jest.fn(x => true),
      permitInfo: {
        [PERMIT_MOBILE_CAR]: {
          entityApproval: {
            isApproved: true,
            documents: {
              test: { name: 'dummy.pdf' },
            },
            showError: false,
          },
        },
      },
      actions: {
        permitInfo: {
          update: jest.fn(),
        },
      },
    };
  });
  it('should render with all props', () => {
    const component = shallow(<EntityApprovalDocs {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('should render with all props with empty currentStep', () => {
    props.currentStep = '';
    const component = shallow(<EntityApprovalDocs {...props} />);
    expect(component).toMatchSnapshot();
  });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  describe('FileUpload', () => {
    it('should handle change event', () => {
      const component = shallow(<EntityApprovalDocs {...props} />);
      const FileUploads = component.find('FileUploads');
      const allProps: any = FileUploads.props();
      const { handleChange } = allProps;
      handleChange(props);
      expect(props.actions.permitInfo.update).toBeCalled();
    });
  });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  // describe('checkbox', () => {
  //   it('should handle checkbox change event', () => {
  //     const component = shallow(<EntityApprovalDocs {...props} />);
  //     const Checkbox = component.find('Memo(Checkbox)');
  //     const allProps: any = Checkbox.props();
  //     const { onChange } = allProps;
  //     onChange(e);
  //     expect(props.handleToggleCheckbox).toBeCalledWith(props);
  //   });
  // });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  describe('BackButton', () => {
    it('should handle button click', () => {
      const component = shallow(<EntityApprovalDocs {...props} />);
      const Form = component.find('LocalizedComponent(Form)');
      const allProps: any = Form.props();
      const { backButton } = allProps;
      backButton.onClick();
      expect(props.onBackClick).toBeCalled();
    });
  });
  // eslint-disable-next-line no-console
  console.log = () => {};
  console.info = () => {};
  console.warn = () => {};
  console.error = () => {};

  describe('SubmitButton', () => {
    it('should handle button click validation success', () => {
      const component = shallow(<EntityApprovalDocs {...props} />);
      const Form = component.find('LocalizedComponent(Form)');
      const allProps: any = Form.props();
      const { submitButton } = allProps;
      submitButton.onClick();
      expect(props.onSubmit).toBeCalled();
    });
    it('should handle button click validation fail', () => {
      props.permitInfo[PERMIT_MOBILE_CAR].entityApproval.isApproved = false;
      props.permitInfo[PERMIT_MOBILE_CAR].entityApproval.showError = true;
      props.validation = jest.fn(x => false);
      const component = shallow(<EntityApprovalDocs {...props} />);
      const Form = component.find('LocalizedComponent(Form)');
      const allProps: any = Form.props();
      const { submitButton } = allProps;
      expect(submitButton.onClick()).toBeFalsy();
    });
  });
});
