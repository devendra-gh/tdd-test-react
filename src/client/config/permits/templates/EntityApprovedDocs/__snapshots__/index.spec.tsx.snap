// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`permits/templates/EntityApprovalDocs should render with all props 1`] = `
<Fragment>
  <Container
    sidebar={
      <Connect(Sidebar)
        currentStep="test"
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "entityApproval.sectionContent",
              ],
              Array [
                "entityApproval.sectionTitle",
              ],
              Array [
                "title.addAttachments",
              ],
              Array [
                "title.termsAndConditions",
              ],
              Array [
                "docs.description",
              ],
              Array [
                "global.termsAndConditions",
              ],
              Array [
                "back",
              ],
              Array [
                "next",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    }
  >
    <ServiceTemplate
      description={
        <div>
          <p />
        </div>
      }
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <FileUploads
      colSpan={4}
      handleChange={[Function]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "entityApproval.sectionContent",
            ],
            Array [
              "entityApproval.sectionTitle",
            ],
            Array [
              "title.addAttachments",
            ],
            Array [
              "title.termsAndConditions",
            ],
            Array [
              "docs.description",
            ],
            Array [
              "global.termsAndConditions",
            ],
            Array [
              "back",
            ],
            Array [
              "next",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      inputGroups={
        Array [
          Object {
            "fields": Array [
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.adpApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.adpApproval",
                "maxSize": 2000000,
                "name": "AdpApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.civilDefenceApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.civilDefenceApproval",
                "maxSize": 2000000,
                "name": "CivilDefenseApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.foodControlApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.foodControlApproval",
                "maxSize": 2000000,
                "name": "FoodControlApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.locationApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.locationApproval",
                "maxSize": 2000000,
                "name": "LocationApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
            ],
            "name": undefined,
            "stateKey": "documents",
          },
        ]
      }
      name="entityApprovalDocs"
      serviceType="mobile-car-tajer-permit-in-abu-dhabi"
      startShowingErrors={false}
      values={
        Object {
          "entityApprovalDocs": Object {
            "test": Object {
              "name": "dummy.pdf",
            },
          },
        }
      }
    />
    <Form.Fieldset
      buttonGroup={false}
      gapSize="default"
      title=""
      twoColumns={false}
    >
      <ServiceTemplate
        description=""
        helpfulBlock={null}
        process={null}
        startLogin={null}
        tables={null}
        title=""
      />
      <Checkbox
        checked={true}
        name="terms"
        onChange={[Function]}
        validateStatus=""
      />
      <LocalizedComponent(Form)
        actions={
          Object {
            "permitInfo": Object {
              "update": [MockFunction],
            },
          }
        }
        backButton={
          Object {
            "alignIcon": "start",
            "label": undefined,
            "onClick": [Function],
            "uiType": "secondary",
            "withArrow": true,
          }
        }
        cancelLink={null}
        currentStep="test"
        handleToggleCheckbox={[MockFunction]}
        help=""
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "entityApproval.sectionContent",
              ],
              Array [
                "entityApproval.sectionTitle",
              ],
              Array [
                "title.addAttachments",
              ],
              Array [
                "title.termsAndConditions",
              ],
              Array [
                "docs.description",
              ],
              Array [
                "global.termsAndConditions",
              ],
              Array [
                "back",
              ],
              Array [
                "next",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        id=""
        onBackClick={[MockFunction]}
        onSubmit={[MockFunction]}
        permitInfo={
          Object {
            "mobile-car-tajer-permit-in-abu-dhabi": Object {
              "entityApproval": Object {
                "documents": Object {
                  "test": Object {
                    "name": "dummy.pdf",
                  },
                },
                "isApproved": true,
                "showError": false,
              },
            },
          }
        }
        serviceType="mobile-car-tajer-permit-in-abu-dhabi"
        submitButton={
          Object {
            "label": undefined,
            "onClick": [Function],
            "withArrow": true,
          }
        }
        validateStatus={null}
        validation={[MockFunction]}
      />
      <div
        style={
          Object {
            "height": 50,
          }
        }
      />
    </Form.Fieldset>
  </Container>
</Fragment>
`;

exports[`permits/templates/EntityApprovalDocs should render with all props with empty currentStep 1`] = `
<Fragment>
  <Container
    sidebar={
      <Connect(Sidebar)
        currentStep=""
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "entityApproval.sectionContent",
              ],
              Array [
                "entityApproval.sectionTitle",
              ],
              Array [
                "title.addAttachments",
              ],
              Array [
                "title.termsAndConditions",
              ],
              Array [
                "docs.description",
              ],
              Array [
                "global.termsAndConditions",
              ],
              Array [
                "back",
              ],
              Array [
                "next",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    }
  >
    <ServiceTemplate
      description={
        <div>
          <p />
        </div>
      }
      helpfulBlock={null}
      process={null}
      startLogin={null}
      tables={null}
      title=""
    />
    <FileUploads
      colSpan={4}
      handleChange={[Function]}
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "entityApproval.sectionContent",
            ],
            Array [
              "entityApproval.sectionTitle",
            ],
            Array [
              "title.addAttachments",
            ],
            Array [
              "title.termsAndConditions",
            ],
            Array [
              "docs.description",
            ],
            Array [
              "global.termsAndConditions",
            ],
            Array [
              "back",
            ],
            Array [
              "next",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      inputGroups={
        Array [
          Object {
            "fields": Array [
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.adpApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.adpApproval",
                "maxSize": 2000000,
                "name": "AdpApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.civilDefenceApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.civilDefenceApproval",
                "maxSize": 2000000,
                "name": "CivilDefenseApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.foodControlApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.foodControlApproval",
                "maxSize": 2000000,
                "name": "FoodControlApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
              Object {
                "accept": Array [
                  "application/pdf",
                ],
                "aria-label": "docs.locationApproval",
                "elementType": "fileUpload",
                "help": "global.uploadHelpPhoto",
                "label": "docs.locationApproval",
                "maxSize": 2000000,
                "name": "LocationApproval",
                "validationConfig": Object {
                  "type": "REQUIRES FILE",
                },
              },
            ],
            "name": undefined,
            "stateKey": "documents",
          },
        ]
      }
      name="entityApprovalDocs"
      serviceType="mobile-car-tajer-permit-in-abu-dhabi"
      startShowingErrors={false}
      values={
        Object {
          "entityApprovalDocs": Object {
            "test": Object {
              "name": "dummy.pdf",
            },
          },
        }
      }
    />
    <Form.Fieldset
      buttonGroup={false}
      gapSize="default"
      title=""
      twoColumns={false}
    >
      <ServiceTemplate
        description=""
        helpfulBlock={null}
        process={null}
        startLogin={null}
        tables={null}
        title=""
      />
      <Checkbox
        checked={true}
        name="terms"
        onChange={[Function]}
        validateStatus=""
      />
      <LocalizedComponent(Form)
        actions={
          Object {
            "permitInfo": Object {
              "update": [MockFunction],
            },
          }
        }
        backButton={
          Object {
            "alignIcon": "start",
            "label": undefined,
            "onClick": [Function],
            "uiType": "secondary",
            "withArrow": true,
          }
        }
        cancelLink={null}
        currentStep=""
        handleToggleCheckbox={[MockFunction]}
        help=""
        i18n={
          [MockFunction] {
            "calls": Array [
              Array [
                "entityApproval.sectionContent",
              ],
              Array [
                "entityApproval.sectionTitle",
              ],
              Array [
                "title.addAttachments",
              ],
              Array [
                "title.termsAndConditions",
              ],
              Array [
                "docs.description",
              ],
              Array [
                "global.termsAndConditions",
              ],
              Array [
                "back",
              ],
              Array [
                "next",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        id=""
        onBackClick={[MockFunction]}
        onSubmit={[MockFunction]}
        permitInfo={
          Object {
            "mobile-car-tajer-permit-in-abu-dhabi": Object {
              "entityApproval": Object {
                "documents": Object {
                  "test": Object {
                    "name": "dummy.pdf",
                  },
                },
                "isApproved": true,
                "showError": false,
              },
            },
          }
        }
        serviceType="mobile-car-tajer-permit-in-abu-dhabi"
        submitButton={
          Object {
            "label": undefined,
            "onClick": [Function],
            "withArrow": true,
          }
        }
        validateStatus={null}
        validation={[MockFunction]}
      />
      <div
        style={
          Object {
            "height": 50,
          }
        }
      />
    </Form.Fieldset>
  </Container>
</Fragment>
`;
